File: src/database/entities/OrderCancel.ts
import { BaseEntity, Column, Entity, JoinColumn, OneToOne, PrimaryGeneratedColumn } from "typeorm";
import { BaseInfoEntity } from "./base/BaseInfoEntity";
import { OrderList } from "./OrderList";

// 주문취소내역
@Entity("ORD_CNCL_DLST")
export class OrderCancel extends BaseEntity {
    // FK
    @OneToOne(type => OrderList, orderList => orderList.getOrderCancel, { lazy: true })
    @JoinColumn([{ name: "SECOMP_ORD_NO", referencedColumnName: "secompOrderNo" }])
    private orderList: OrderList;

    // PK
    @PrimaryGeneratedColumn()
    private id: number;

    // 취소증권사주문번호
    @Column({
        name: "CNCL_SECOMP_ORD_NO",
        length: 20,
    })
    private cancelSecompOrderNo: string;

    // 공통정보
    @Column((type: any) => BaseInfoEntity)
    private baseInfo: BaseInfoEntity;

    public static createOrderCancel = (orderList: OrderList, cancelSecompOrderNo: string): OrderCancel => {
        const orderCancel: OrderCancel = new OrderCancel();

        orderCancel.orderList = orderList;
        orderCancel.cancelSecompOrderNo = cancelSecompOrderNo;

        return orderCancel;
    };

    public getOrderList = () => {
        return this.orderList;
    };
}

File: src/main/java/com/example/kisautotrading/domain/order/domain/OrderCancel.java
package com.example.kisautotrading.domain.order.domain;

import com.example.kisautotrading.global.common.entity.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

// 주문취소내역
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity(name = "ORD_CNCL_DLST")
public class OrderCancel extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_list_id", nullable = false)
    private OrderList orderList;

    @Column(name = "CNCL_SECOMP_ORD_NO")
    private String cancelSecompOrderNo; // 취소증권사주문번호

    @Builder
    public OrderCancel(OrderList orderList, String cancelSecompOrderNo) {
        this.orderList = orderList;
        this.cancelSecompOrderNo = cancelSecompOrderNo;
    }
}
